### ISSUE TITLE
Enhancement Request: Change `aqt tool` to not require version and arch (--latest)

### ISSUE THREAD

**Issue by jmarrec** [miurahr/aqtinstall#282]
I can see that `tools_ifw` always breaks because Qt will remove any previous versions (there's only one). It seems we could just parse the Updates.xml and get the latest one if the version isn't actually supplied.

Also, it seems like error reporting needs to be improved. Even if the tool isn't found, the "Installation finished" is issued.

I will try to provide a PR for review

**Comment by ddalcino** [comment]
I have some notes on the `aqt tool` functionality, based on reading all the `Updates.xml` files in the tool folders at https://download.qt.io/online/qtsdkrepository/. This issue might not be the right place to put these notes, but should be helpful to anyone trying to work on the `aqt tool` feature.

I ran a script that extracts all the **'Updates > PackageUpdate > Version'** tags in the `Updates.xml` files for the tools aqt should be able to install, and tried to use `semantic_version.Version` to parse the versions. Most of them parse without any problem. The following packages contain versions that fail to parse:

* https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_vcredist/Updates.xml contains the following packages, with version strings that fail to parse:
    * qt.tools.vcredist: 2011-03-03-1
    * qt.tools.vcredist_msvc2015_x86: 2017-02-01-0
    * qt.tools.vcredist_msvc2013_x64: 2017-3-22-0
    * qt.tools.vcredist_msvc2015_x64: 2017-03-22-0
    * qt.tools.vcredist_msvc2013_x86: 2014-30-12-1
    * qt.tools.vcredist_msvc2017_x64: 2019-02-13-1
    * qt.tools.vcredist_msvc2017_x86: 2019-02-13-1
    * qt.tools.vcredist_64: 2020-03-18
    * qt.tools.vcredist_msvc2019_x64: 2020-05-19-1
    * qt.tools.vcredist_msvc2019_x86: 2019-05-19-1
* https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_qtcreator/Updates.xml contains this package:
    * qt.tools.windows_kits_debuggers: 2019-11-07-0
* https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_conan/Updates.xml contains this package:
    * qt.tools.conan: 1.33-202102101246
* https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/tools_conan/Updates.xml contains this package:
    * qt.tools.conan: 1.33-202102101246
* https://download.qt.io/online/qtsdkrepository/mac_x64/desktop/tools_conan/Updates.xml contains this package:
    * qt.tools.conan: 1.33-202102101246

**TLDR:** The `Updates.xml` files in `tools_vcredist` and `tools_qtcreator` on Windows contain Version tags that contain dates, not versions. Also, `tools_conan` has packages on all three platforms whose versions cannot be parsed because they are missing a patch number.

**Conclusion:** Whoever writes the code that reads the Version tags needs to account for:
* versions without patch numbers, and
* dates instead of versions

**Comment by miurahr** [comment]
Related #233

**Comment by miurahr** [comment]
PR #299 is a quick hack to realize the feature.

**Comment by ddalcino** [comment]
I have a question about the current and planned functionality of `aqt tool`:

Currently, ToolArchive hardcodes the `target` parameter to `desktop`: https://github.com/miurahr/aqtinstall/blob/61913c21285c149bc8c0a5f2aa5a04c82cd450ec/aqt/archives.py#L789-L796
This makes it impossible to install tools for android, ios, or winrt. Is there a reason for this? Do we have any plans to make those tools available for installation? I have not tried it yet, but it looks like it would be a very easy change to make.

**Comment by miurahr** [comment]
> This makes it impossible to install tools for android, ios, or winrt. Is there a reason for this? Do we have any plans to make those tools available for installation? I have not tried it yet, but it looks like it would be a very easy change to make.

Because this is quick hack and proof of concept. You are welcome to contribute for a complete solution.

**Comment by miurahr** [comment]
> This makes it impossible to install tools for android, ios, or winrt. Is there a reason for this? Do we have any plans to make those tools available for installation? I have not tried it yet, but it looks like it would be a very easy change to make.

It would be better accept target "desktop", "andorid" and others for command line argument.

**Comment by miurahr** [comment]
#299 takes an approach that it does not check version.
#303 checks version with spec.
@ddalcino  #303 will supersede  #299. Is it right?

**Comment by miurahr** [comment]
Oh, #303 request to merge to #299.

**Comment by ddalcino** [comment]
I closed #303; I don't think it's really as useful as I thought it would be.

**Comment by jmarrec** [comment]
Fyi, I know I disappeared suddenly, but I had a PR almost ready. I was just waiting for #287 (the cosmetic part) to be accepted first before submitting it, but left for vacation in the meantime.
That subsequent PR just makes the two args optional and nothing else. I can submit it in a few days

**Comment by miurahr** [comment]
Now #299 is implement the feature #282 requested and ready to merge.

**Comment by miurahr** [comment]
v2.0.0b5 out with the improvement.
